### ISSUE TITLE
Unsafe statement written to the binary log

### ISSUE THREAD

**Issue by RealRancor** [owncloud/music#289]
### Steps to reproduce
1. Enabling music app 0.3.4
2. Watch mysql error logfile

### Expected behaviour
No warnings about unsafe statement written to the binary log should be shown in the mysql error log

### Actual behaviour
Warnings about unsafe statement written to the binary log are shown in the mysql error log

### Server configuration
**Operating system**: Debian wheezy

**Web server:** NginX 1.6.x

**Database:** MariaDB 10.0.x

**PHP version:** 5.5.x

**ownCloud version:** 7.0.2

**Updated from an older ownCloud or fresh install:** Updated from OC6

**List of activated apps:** Default apps + mozilla_sync + music

**The content of config/config.php:**

```
<?php

define("DEBUG", true);

$CONFIG = array (
  'instanceid' => 'xxx',
  'passwordsalt' => 'xxx',
  'datadirectory' => '/var/www/owncloud/data',
  'dbtype' => 'mysql',
  'version' => '7.0.2.1',
  'dbname' => 'xxx',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'xxx',
  'dbpassword' => 'xxx',
  'installed' => true,
  'forcessl' => true,
  'maintenance' => false,
  'mail_smtpmode' => 'php',
  'trusted_domains' =>
  array (
    0 => 'example.com',
  ),
  'loglevel' => '0',
);
```

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

#### mysql error log
```
 [140912 11:00:05] 140912 11:00:05 [Warning] Unsafe statement written to the binary log using statement format since BINLOG_FORMAT = STATEMENT. Statements writing to a table with an auto-increment column after selecting from another table are unsafe because the order in which rows are retrieved determines what (if any) rows will be written. This order cannot be predicted and may differ on master and the slave. Statement: DELETE FROM `oc_music_tracks` WHERE `file_id` NOT IN (SELECT `fileid` FROM `oc_filecache`)

 [140912 19:42:02] 140912 19:42:02 [Warning] Unsafe statement written to the binary log using statement format since BINLOG_FORMAT = STATEMENT. Statements writing to a table with an auto-increment column after selecting from another table are unsafe because the order in which rows are retrieved determines what (if any) rows will be written. This order cannot be predicted and may differ on master and the slave. Statement: DELETE FROM `oc_music_albums` WHERE `id` NOT IN (SELECT `album_id` FROM `oc_music_tracks` GROUP BY `album_id`)

 [140912 01:42:02] 140912  1:42:02 [Warning] Unsafe statement written to the binary log using statement format since BINLOG_FORMAT = STATEMENT. Statements writing to a table with an auto-increment column after selecting from another table are unsafe because the order in which rows are retrieved determines what (if any) rows will be written. This order cannot be predicted and may differ on master and the slave. Statement: UPDATE `oc_music_albums` SET `cover_file_id` = NULL WHERE `cover_file_id` IS NOT NULL AND `cover_file_id` NOT IN (SELECT `fileid` FROM `oc_filecache`)

 [140912 05:56:02] 140912  5:56:02 [Warning] Unsafe statement written to the binary log using statement format since BINLOG_FORMAT = STATEMENT. Statements writing to a table with an auto-increment column after selecting from another table are unsafe because the order in which rows are retrieved determines what (if any) rows will be written. This order cannot be predicted and may differ on master and the slave. Statement: DELETE FROM `oc_music_artists` WHERE `id` NOT IN (SELECT `artist_id` FROM `oc_music_album_artists` GROUP BY `artist_id`)
```

**Comment by MorrisJobke** [comment]
@RealRancor Is this fixed now?

**Comment by RealRancor** [comment]
Will try 0.3.7 and give feedback / close the issue if fixed. Thanks for the reminder.

**Comment by RealRancor** [comment]
Ah, nevermind. #347 was already merged for 0.3.6. Just checked the mysql error log and its empty so this is fixed from my PoV.
